draft-issuance.md

```diff
@@ -0,0 +1,161 @@
+
+Additionally, the current Bitcoin-style issuance does not take into account the current balance of `ZsfBalanceAfter(h)`. If [#draft-zsf]_ were to activate without a change to the issuance mechanism, then some funds would never be disbursed after they are deposited back into the ZSF.
+
+In summary, by introducing a smoother emissions curve, we:
+- maintain the economic viability of Zcash
+- provide clarity on reward distribution
+- take into account deposits into the ZSF
+- enhance Zcash's stability as the network evolves.
+
+# Requirements
+
+Smoothing the issuance curve is possible using an exponential decay formula that
+satisfies the following requirements:
+
+## Issuance Requirements
+
+1. The issuance can be summarised into a reasonably simple explanation
+2. Block subsidies approximate a continuous function
+3. If there are funds in the ZSF, then the block subsidy must be non-zero, preventing any final “unmined” zatoshis
+4. For any 4 year period, all paid out block subsidies are approximately equal to half of the ZSF at the beginning of that 4 year period, if there are no deposits into the ZSF during those 4 years
+TODO daira: add a requirement that makes the initial total issuance match the previous total issuance  
+5. This functionality MUST be introduced as part of a network upgrade
+
+
+# Specification
+
+## Constants
+
+Define constants:
+
+“`BLOCK_SUBSIDY_FRACTION`” = 41 / 100,000,000 or `0.00000041`
+
+"`NUMBER_OF_BLOCKS_IN_4_YEARS`" = `(365.25 * 24 * 60 * 60/75 * 4)`, or `1_683_072`
+
+## Issuance Calculation
+
+Given the block height `h` define a function **BlockSubsidy(h)**, such that:
+
+**BlockSubsidy(h)** = Block subsidy for a given `h`, that satisfies above requirements.
+
+Using an exponential decay function for **BlockSubsidy** satisfies requirements **R1** and **R2** above:
+
+`BlockSubsidy(h) = BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1)`
+
+Finally, to satisfy **R3** above we always round up to the next zatoshi.
+
+`BlockSubsidy(h) = ceiling(BLOCK_SUBSIDY_FRACTION * ZsfBalanceAfter(h - 1))`
+
+# Rationale
+
+## `BLOCK_SUBSIDY_FRACTION`
+
+The value `41 / 100_000_000` satisfies the approximation:
+
+`(1 - BLOCK_SUBSIDY_FRACTION)^NUMBER_OF_BLOCKS_IN_4_YEARS ≈ 0.5`
+
+Meaning after a period of 4 years around half of `ZSF_BALANCE` will be paid out
+as block subsidies, thus satisfying **R4**.
+
+## Visualization of the Smoothed Curve
+
+The following graph, taken from the ECC blog post, illustrates the smoothed curve. Note that depending on when the network upgrade takes place the disbursement may temporarily _increase_.
```

Comment:
We want to replace this with an accurate graph.

Reply:
Note: the blog post used [this repository for generating the plots](https://github.com/nathan-at-least/zec-issuance-plots).

One approach I think would be cool would be to copy that code directly into the ZIPs source and update the build system so that the plots are generated by the ZIPs build process. This way changes to the ZIP contents and the literaly plots generated could be within the scope of single PRs.

If anyone likes that approach and wants to work on it, I can chip in!

Reply:
@nathan-at-least I've uploaded my code here: https://github.com/eigerco/zsf-simulator

Reply:
note: @tomekpiotrowski is going to work on the graph tomorrow